packages/medusa/src/repositories/product.ts

```diff
@@ -42,6 +429,134 @@ export const ProductRepository = dataSource.getRepository(Product).extend({
     return this.findByIds(productIds)
   },
 
+  async getFreeTextSearchResultsAndCount(
+    q: string,
+    options: FindWithoutRelationsOptions = { where: {} },
+    relations: string[] = []
+  ): Promise<[Product[], number]> {
+    const productAlias = "product"
+    const pricesAlias = "prices"
+    const variantsAlias = "variants"
+    const collectionAlias = "collection"
+    const tagsAlias = "tags"
+
+    const tags = options.where.tags
+    delete options.where.tags
+
+    const price_lists = options.where.price_list_id
+    delete options.where.price_list_id
+
+    const sales_channels = options.where.sales_channel_id
+    delete options.where.sales_channel_id
+
+    const discount_condition_id = options.where.discount_condition_id
+    delete options.where.discount_condition_id
+
+    const categoriesQuery = options.where.categories
+    delete options.where.categories
+
+    const cleanedOptions = this._cleanOptions(options)
```

Comment:
question: If we want a utility to clean the options, wouldn't it make sense to include all of the above?